tst/com/amazon/corretto/crypto/provider/test/AESGenerativeTest.java

```diff
@@ -63,9 +65,35 @@ public void teardown() {
     amzCipherDecrypt = null;
   }
 
+  @Test
+  public void testEncrypt91() throws Exception {
+    // The implementation of javax.crypto.CipherSpi::bufferCrypt in JDK 10
+    // has an issue that has been resolved in the following PR:
+    // https://github.com/openjdk/jdk/commit/c3a97b27e244f97c80a14388aea4fc425006a87e
+    // We disable this test for JDK 10.
+
+    // To replicate the error for other JDKs, copy-paste the buggy implementation into
+    // AesGcmSpi.java
+    // and run `./gradlew cmake_clean single_test
+    // -DSINGLE_TEST=com.amazon.corretto.crypto.provider.test.AESGenerativeTest`
+    // The buggy implementation is available here:
+    // https://github.com/openjdk/jdk/blob/f98aad58dea1d0b818706215166bcbbc349d1c6d/src/java.base/share/classes/javax/crypto/CipherSpi.java#L736-L857
+    assumeTrue(getJavaVersion() != 10);
+    try {
+      testEncrypt(91, 10);
+    } catch (Throwable t) {
+      throw new AssertionError("Seed: 91", t);
+    }
+  }
+
   @Test
   public void testEncryptRandomly() throws Exception {
     for (int i = 0; i < 100; i++) {
+      // when i == 91, the test case fails for JDK 10. This case is tested in another method:
+      // testEncrypt91
+      if (i == 91) {
```

Comment:
```suggestion
      if (i == 91 && getJavaVersion() == 10) {
```

nit: suggest only skipping this when we _need_ to.

Reply:
For `i == 91`,  there is another method `testEncrypt91()`  for testing this case.